### PR TITLE
replace insecure-key support with authorized-key support

### PR DESCRIPTION
**Pull request by bukzor**
I think this fills all the use cases of the insecure-key, plus some, and is overall more secure.

### CODE REVIEW

File: test/test.sh [hunk 1]
```diff
@@ -8,15 +8,19 @@ function ok()
 
 function fail()
 {
-	echo "  FAIL"
```
Comment (endersonmaia) [inline test/test.sh]: This changes could be in another PR, since it's not related exclusively with the authorized_keys support, I think.

File: Changelog.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+## -bukzor0 (not planned for release)
```
Comment (endersonmaia) [inline Changelog.md]: Let this changes for the maintainer.

File: Makefile [hunk 3]
```diff
@@ -1,5 +1,5 @@
 NAME = phusion/baseimage
-VERSION = 0.9.16
+VERSION = 0.9.16-bukzor0
```
Comment (endersonmaia) [inline Makefile]: This should be defined by the maitainers, since it could go to the next or further release.

**Comment by endersonmaia** [comment]
@bukzor this PR is fine, but I don't know if @FooBarWidget would like to make insecure_key and authorized_keys optional, or keep one of them.

I suggest removing all changes that are not related with the authorized_keys feature, and if it's the case, open a new PR for each os these other changes.

That's my 2cents, HTH.

File: test/test.sh (hunk 1 above)
Comment (bukzor) [inline test/test.sh]: it came up during testing this change

File: Changelog.md (hunk 2 above)
Comment (bukzor) [inline Changelog.md]: he'll want the change notes. of course he will change the version

**Comment by Theaxiom** [comment]
@bukzor Is there an existing issue that this PR is in reference to? If not, please open a new issue which explains what you are trying to accomplish with these changes. Thank you!

This PR also has merge conflicts, please kindly resolve those conflicts. :)
